fontbakery-check-ttf.py

```diff
@@ -685,6 +720,149 @@ def ttf_file(f):
+                    "ExtraLightItalic": "ExtraLight",
+                    "LightItalic": "Light",
+                    "MediumItalic": "Medium",
+                    "SemiBoldItalic": "SemiBold",
+                    "ThinItalic": "Thin"}
+
+      if value in onlyWeight.keys():
+        return onlyWeight[value]
+      else:
+        return value
+
+    filename = os.path.split(font_file)[1]
+    filename_base = os.path.splitext(filename)[0]
+    fname, style = filename_base.split('-')
+    fname_with_spaces = family_with_spaces(fname)
+    style_with_spaces = stylename_with_spaces(style)
+    only_weight = get_only_weight(style)
+
+    for name in font['name'].names:
+      string = name.string.decode(name.getEncoding()).strip()
+      nameid = name.nameID
+      plat = name.platformID
+      expected_value = ""
+      expected_values = None
+
+      if nameid == NAMEID_FONT_FAMILY_NAME:
+        if plat == PLATFORM_ID_MACHINTOSH:
+          expected_value = fname_with_spaces
+        elif plat == PLATFORM_ID_WINDOWS:
+          if style in ['Regular',
+                       'Italic',
+                       'Bold',
+                       'Bold Italic']:
+            expected_value = fname_with_spaces
+          else:
+            expected_value = " ".join([fname_with_spaces,
+                                       only_weight]).strip()
+        else:
+          fb.error(("Font should not have a "
+                    "[{}:{}] entry!").format(NAMEID_STR[nameid],
+                                             PLATID_STR[plat]))
+          continue
+      elif nameid == NAMEID_FONT_SUBFAMILY_NAME:
+        if style_with_spaces not in STYLE_NAMES:
+          fb.error(("Style name '{}' inferred from filename"
+                    " is not canonical."
+                    " Valid options are: {}").format(style_with_spaces,
+                                                     STYLE_NAMES))
+          continue
+
+        if plat == PLATFORM_ID_MACHINTOSH:
+          expected_value = style_with_spaces
+
+        elif plat == PLATFORM_ID_WINDOWS:
+          if style_with_spaces in ["Bold", "Bold Italic"]:
+            expected_value = style_with_spaces
+          else:
+            if "Italic" in style:
+              expected_value = "Italic"
+            else:
+              expected_value = "Regular"
+
+      elif name.nameID == NAMEID_FULL_FONT_NAME:
+        if plat == PLATFORM_ID_MACHINTOSH:
+          if style == "Regular":
+            expected_value = fname_with_spaces
+          else:
+            expected_value = "{} {}".format(fname_with_spaces,
+                                            style_with_spaces)
+        elif plat == PLATFORM_ID_WINDOWS:
+          expected_values = ["{}-{}".format(fname,
+                                        style),
+                             "{} {}".format(fname_with_spaces,
+                                            style_with_spaces)]
+
+      elif name.nameID == NAMEID_POSTSCRIPT_NAME:
```

Comment:
This is the last name.nameID elif, but there need to be elif's for NAMEID_TYPOGRAPHIC_FAMILY_NAME and NAMEID_TYPOGRAPHIC_SUBFAMILY_NAME